### DISCUSSION TITLE
Supabase returns null data (RLS Off)

### DISCUSSION THREAD

**Discussion by Ryurcia** [supabase/supabase#22221] [Questions] [answered]
Im currently using the supabase api to return my user data

`const {data} = await supabase.from('Users').select('*').eq('user_id',userId).single()`

It would sometimes return my data and sometimes it just returns null. Ive alreadt tried turning RLS off but it would still sometimes give me a null response.

**Comment by GaryAustin1** [comment, marked as answer]
Are you checking for errors?  .single() requires exactly 1 row.  0 or 2 will return null, but with an error.

**Reply by Ryurcia** [reply]
Fixed it. I needed to use .limit(1) before .single()
